lib/discourse_automation/scriptable.rb

```diff
@@ -211,19 +211,32 @@ def self.send_pm(
           pm[:prefers_encrypt] = prefers_encrypt
           DiscourseAutomation::PendingPm.create!(pm)
         else
-          if sender = User.find_by(username: sender)
-            post_created = false
-            pm = pm.merge(archetype: Archetype.private_message)
+          sender = User.find_by(username: sender)
+          if !sender
+            Rails.logger.warn "[discourse-automation] Did not send PM #{pm[:title]} - sender does not exist: `#{sender}`"
+            return
+          end
 
-            if prefers_encrypt
-              pm[:target_usernames] = (pm[:target_usernames] || []).join(",")
-              post_created = EncryptedPostCreator.new(sender, pm).create
-            end
+          if (pm[:target_usernames] || []).empty?
+            Rails.logger.warn "[discourse-automation] Did not send PM - no target usernames"
+            return
+          end
 
-            PostCreator.new(sender, pm).create! if !post_created
-          else
-            Rails.logger.warn "[discourse-automation] Couldn’t send PM to user with username: `#{sender}`."
+          no_user = pm[:target_usernames].filter { |u| !User.find_by(username: u) }
```

Comment:
This is going to trigger one query for each username we want to check. I think you can do something like `User.where(username: pm[:target_usernames]).pluck(:username_lower).uniq.length`